### ISSUE TITLE
What do you mean by 'the model is being trained'?

### ISSUE THREAD

**Issue by bluddy** [nashory/pggan-pytorch#12]
Does this code work? Are you able to attain high-res quality images, as in the paper?

**Comment by nashory** [comment]
yeah, I guess it's been a quite long time since my last update.
The code is working properly until it reaches 256 resolution.
I did not try with CelebA-HQ yet, but it should be working as well :)
Most bugs have been fixed already, so please feel free to use this.
(I will update the 256x256 results soon. too busy these days!)

**Comment by bluddy** [comment]
Am I correct in saying this is still very slow? I've had it running for 3 days on a Titan XP, and my resulting faces are somewhat... monstrous. I'm using default values, btw.

**Comment by nashory** [comment]
my results are shown in README.
what do you mean monstrous? Is it far from mine?
could you show me what you've got?
And yes, the training is slow. The authors trained almost a week as well.

**Comment by bluddy** [comment]
Well, some of the images are pretty good, but even after almost a week, many of the images come out terrible -- pieces don't fit together, or are really smudged. Did NVIDIA hand-pick the images in the paper to look that good?

**Comment by nashory** [comment]
I've trained to reproduce the issue with default settings.
The current result is like below, and it took around 3 days using P40 single GPU.
Yes, I see some images look smashed terribly. (but not that much, though.)

My hypothesis about this is that:
+ 1. NVIDIA authors hand-picked good looking images? (not sure)
+ 2. It might be working well if using easier domain? (performing face detection and crop on CelebA, or using Anime dataset?)
+ 3. There is something wrong in my code? (possibly not... checked more than thousands of times)
+ 4. Wrong optimization parameter? (the result changes significantly depending on the adam optimizer beta1/beta2 values.)

I think we need to go through this issue based on the hypothesis above.
what do you think?

### my result so far.
![2017-12-29 10 31 31](https://user-images.githubusercontent.com/17468992/34427386-ebbd83dc-ec84-11e7-900d-dd3d9b7d34d7.png)
![2017-12-29 10 31 54](https://user-images.githubusercontent.com/17468992/34427388-ecf06166-ec84-11e7-9d6d-83dbf69a157d.png)

**Comment by bluddy** [comment]
Yeah I'm getting the same kind of results. Some areas are 'smushed', but more importantly, the network doesn't seem to be able to combine the pieces from the features in a seamless way, causing the uncanny valley / 'Frankenstein' effect. This problem was noticeable at lower resolution levels as well, but I was hoping it would get better, though it didn't.

Maybe the algorithm needs to run longer on lower resolutions to work out those kinks before we go up in resolution. Perhaps once those patterns are there, they can't be eliminated, since the lower layers have already been formed for the most part.

**Comment by nashory** [comment]
Have you tried to use wasserstein loss by any chance?

**Comment by AnushaManila** [comment]
By loading the CelebA HQ dataset and using the same code, I'm getting similar results to NVIDIA.
The model works fine and the sample result that generated at 512X512 resolution is attached.
![11_celebhq_512](https://user-images.githubusercontent.com/12676003/35472279-bfe01f3c-036c-11e8-9bf5-05d48afa21d9.png)

**Comment by nashory** [comment]
At the moment, the dataloader resize the input image on cpu in real time. If resize all images in advance, the training time would be decreased significantly. By the way, the result now looks pretty good :)

**Comment by AnushaManila** [comment]
Thank you for sharing this code in pytorch!
I have cited this GitHub in my thesis.
Pggan experiments are a part of my thesis work, have not planned more than this for now:-)
